build_linux_nvidia.sh

```diff
@@ -0,0 +1,381 @@
+  fi
+
+  source ${PYTHON_VENV_DIR}/bin/activate
+  python3 get-pip.py
+  pip3 install testresources
+  pip3 install --upgrade setuptools wheel
+
+  # Latest PIL is not compatible with mmcv=1.4.1
+  pip install Pillow==7.0.0
+
+  if [[ "$ARCH" == aarch64 ]]
+  then
+    # protofbuf on jetson is quite old - must be upgraded
+    pip3 install --upgrade protobuf
+    # Install numpy 1.19.4 as newer versions might give "Illegal instruction (core dumped)" on Jetson
+    # TODO Numpy might be installed per default so we should not remove it.
+    pip3 install numpy==1.19.4
+  else
+    pip3 install numpy
+  fi
+  pip3 install opencv-python
+  pip3 install matplotlib
+
+  # pytorch, torchvision, torchaudio
+  if [[ "$ARCH" == aarch64 ]]
+  then
+    # pytorch
+    wget https://nvidia.box.com/shared/static/fjtbno0vpo676a25cgvuqc1wty0fkkg6.whl -O torch-1.10.0-cp36-cp36m-linux_aarch64.whl
+    pip3 install torch-1.10.0-cp36-cp36m-linux_aarch64.whl
+    # torchvision
+    sudo apt-get install libjpeg-dev zlib1g-dev libpython3-dev libavcodec-dev libavformat-dev libswscale-dev -y
+    sudo rm -r torchvision
+    git clone --branch v0.11.1 https://github.com/pytorch/vision torchvision
+    cd torchvision
+    export BUILD_VERSION=0.11.1  # where 0.x.0 is the torchvision version
+    python3 setup.py install
+    cd ../
+    # torchaudio
+    #sudo apt-get install -y sox libsox-dev libsox-fmt-all
+    #sudo rm -r torchaudio
+    #git clone -b v0.10.0 https://github.com/pytorch/audio torchaudio
+    #cd torchaudio
+    #git submodule update --init --recursive
+    #python3 setup.py install
+    #cd ../
+    # mmcv
+    pip3 uninstall mmcv-full
+    pip3 install mmcv-full==1.4.1 -f https://download.openmmlab.com/mmcv/dist/cu102/torch1.10.0/index.html
+  else
+    pip3 install torch==1.10.0+cu113 torchvision==0.11.1+cu113 torchaudio==0.10.0+cu113 -f https://download.pytorch.org/whl/cu113/torch_stable.html
+    # mmcv
+    pip3 uninstall mmcv-full
+    pip3 install mmcv-full==1.4.1 -f https://download.openmmlab.com/mmcv/dist/cu113/torch1.10.0/index.html
+  fi
+
+  # deactivate python venv again
+  deactivate
+}
+
+pplcv() {
+  ## ppl.cv
+  cd ${WORKING_DIR}
+  echo_blue "checking out '${PPLCV_DIR}' pkg..."
+  if [ -d "${PPLCV_DIR}" ]; then
+      echo_green "Already exists! Checking out the requested version..."
+  else
+      git clone https://github.com/openppl-public/ppl.cv.git ${PPLCV_DIR}
+  fi
+  cd ${PPLCV_DIR}
+  git pull
+  git checkout tags/v${PPLCV_VER}
+  mkdir build -p && cd build
+  cmake -DHPCC_USE_CUDA=ON -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX} .. && make -j${PROC_NUM} && sudo make install
+  sudo ldconfig
+
+  # TODO pack prebuild as tar.gz file
+  #cd ..
+  #tar -zcvf ${WORKING_DIR}/pplcv_${PPLCV_VER}_cuda-${ARCH}-build.tar.gz build/
+}
+
+mmdeploy(){
+  ## mmdeploy SDK
+  cd ${WORKING_DIR}
+  echo_blue "checking out '${MMDEPLOY_DIR}' pkg..."
+  if [ -d "${MMDEPLOY_DIR}" ]; then
```

Comment:
Since `MMDEPLOY_DIR=${WORKING_DIR}/MMDeploy`, we have to clone it again even when the script is inside the repo dir.

Reply:
@grimoire Yes, you are right. My script was developed with a root working directory, containing MMDeploy. But in our case, the root directory will always be MMDeploy. Therefore we do not need to check out a specific version anymore. I will remove this functionality.